integration/spark/app/src/test/java/io/openlineage/spark/agent/GoogleCloudIntegrationTest.java

```diff
@@ -0,0 +1,194 @@
+    Configuration configuration = new Configuration();
+    configuration.logLevel(Level.ERROR);
+    mockServer = ClientAndServer.startClientAndServer(configuration, MOCKSERVER_PORT);
+    mockServer
+        .when(request("/api/v1/lineage"))
+        .respond(org.mockserver.model.HttpResponse.response().withStatusCode(201));
+  }
+
+  @AfterAll
+  @SneakyThrows
+  public static void afterAll() {
+    SparkSession$.MODULE$.cleanupAnyExistingSession();
+    mockServer.stop();
+  }
+
+  @BeforeEach
+  @SneakyThrows
+  public void beforeEach() {
+    spark =
+        SparkSession.builder()
+            .master("local[*]")
+            .appName("GoogleCloudIntegrationTest")
+            .config("spark.driver.host", LOCAL_IP)
+            .config("spark.driver.bindAddress", LOCAL_IP)
+            .config("spark.sql.shuffle.partitions", 1)
+            .config("spark.sql.warehouse.dir", "file:/tmp/iceberg/")
+            .config("spark.driver.extraJavaOptions", "-Dderby.system.home=/tmp/gctest")
+            .config("spark.openlineage.transport.type", "http")
+            .config(
+                "spark.openlineage.transport.url",
+                "http://localhost:" + mockServer.getPort() + "/api/v1/namespaces/gc-namespace")
+            .config("spark.openlineage.facets.disabled", "spark_unknown;spark.logicalPlan")
+            .config("spark.openlineage.debugFacet", "enabled")
+            .config("parentProject", "openlineage-ci")
+            .config("credentialsFile", "build/gcloud/gcloud-service-key.json")
+            .config("temporaryGcsBucket", "openlineage-spark-bigquery-integration")
+            .config(
+                "spark.hadoop.fs.AbstractFileSystem.gs.impl",
+                "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFS")
+            .config(
+                "spark.hadoop.google.cloud.auth.service.account.json.keyfile",
+                "build/gcloud/gcloud-service-key.json")
+            .config(
+                "spark.hadoop.fs.gs.impl", "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem")
+            .config("spark.hadoop.google.cloud.auth.service.account.enable", "true")
+            .config("spark.extraListeners", OpenLineageSparkListener.class.getName())
+            .getOrCreate();
+
+    spark.sparkContext().setLogLevel("WARN");
+  }
+
+  @Test
+  @EnabledIfEnvironmentVariable(named = "CI", matches = "true")
+  @EnabledIfSystemProperty(named = SPARK_VERSION, matches = SPARK_3) // Spark version >= 3.*
+  void testReadAndWriteFromBigquery() {
+    String PROJECT_ID = "openlineage-ci";
+    String DATASET_ID = "airflow_integration";
+    String versionName = System.getProperty(SPARK_VERSION).replace(".", "_");
+
+    String source_table = String.format("%s.%s.%s_source", PROJECT_ID, DATASET_ID, versionName);
+    String target_table = String.format("%s.%s.%s_target", PROJECT_ID, DATASET_ID, versionName);
+
+    spark.sparkContext().setLogLevel("info");
+
+    // TODO: would love to avoid this in future
```

Comment:
Spark SQL `CREATE TABLE IF NOT EXISTS`?

Reply:
we tried this quickly some time ago, but it didn't work for BigQuery table. This changes just moves code test from python, so I would leave this as is.